custom/echis_reports/ucr/data_sources/leprosy_datasource.json

```diff
@@ -0,0 +1,3682 @@
+{
+    "domains": [
+        "fmoh-echis",
+        "fmoh-echis-staging",
+        "fmoh-echis-link-test"
+    ],
+    "server_environment": [
+        "production",
+        "echis"
+    ],
+    "config": {
+        "configured_indicators": [
+          {
+            "comment": null,
+            "create_index": true,
+            "display_name": "quarterly_report_date",
+            "datatype": "string",
+            "expression": {
+              "datatype": "string",
+              "type": "property_name",
+              "property_name": "hew_screening_report_quarterly"
+            },
+            "transform": {},
+            "is_primary_key": false,
+            "is_nullable": true,
+            "type": "expression",
+            "column_id": "quarterly_report_date"
+          },
+          {
+            "comment": null,
+            "create_index": true,
+            "display_name": "quarterly_date",
+            "datatype": "date",
+            "expression": {
+              "datatype": null,
+              "type": "property_name",
+              "property_name": "hew_screening_date"
+            },
+            "transform": {},
+            "is_primary_key": false,
+            "is_nullable": true,
+            "type": "expression",
+            "column_id": "quarterly_date"
+          },
+          {
+            "type": "expression",
+            "column_id": "region",
+            "datatype": "string",
+            "display_name": "region",
+            "expression": {
+              "test": {
+                "operator": "eq",
+                "expression": {
+                  "type": "root_doc",
+                  "expression": {
+                    "type": "property_name",
+                    "property_name": "owner_id",
+                    "datatype": null
+                  }
+                },
+                "type": "boolean_expression",
+                "comment": null,
+                "property_value": "-"
+              },
+              "type": "conditional",
+              "expression_if_true": {
+                "value_expression": {
+                  "value_expression": {
+                    "type": "root_doc",
+                    "expression": {
+                      "type": "property_name",
+                      "property_name": "name",
+                      "datatype": null
+                    }
+                  },
+                  "type": "related_doc",
+                  "related_doc_type": "Location",
+                  "doc_id_expression": {
```

Comment:
@avazirna do you have concerns with this? It looks like this is going to fetch multiple locations for every document (and does it multiple times in this same datasource). 

@andyasne have you confirmed the performance of this datasource against a sample document, and made sure that it will rebuild in a reasonable amount of time?

Reply:
@proteusvacuum we tested it on the cloud and It did finish building. The reason why I did those calculations is the case type that the report is working with is an extension case in which will not have a  owner Id, So we fetch the parent and get the owner Id location and from that find the full path of the location from the root. 
The other solution is Attaching location information as case property on cases, this is creating problems when cases are reassigned to other locations. reassigning cases will only update the owner Id not the cases properties.

Reply:
@avazirna Modified Leprosy location fields , Removed performance heavy calculations from the datasource and the report.